Creating a blog app with reactive + MySQL doesn't compile

##### **Overview of the issue**

If you create a simple blog monolith with reactive and MySQL, it fails to compile.

```
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ blog ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 68 source files to /Users/mraible/dev/choose-mono/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/mraible/dev/choose-mono/src/main/java/com/jhipster/demo/blog/repository/BlogRepositoryInternalImpl.java:[5,51] cannot find symbol
  symbol:   class UserRowMapper
  location: package com.jhipster.demo.blog.repository.rowmapper
[ERROR] /Users/mraible/dev/choose-mono/src/main/java/com/jhipster/demo/blog/repository/BlogRepositoryInternalImpl.java:[40,19] cannot find symbol
  symbol:   class UserRowMapper
  location: class com.jhipster.demo.blog.repository.BlogRepositoryInternalImpl
[ERROR] /Users/mraible/dev/choose-mono/src/main/java/com/jhipster/demo/blog/repository/BlogRepositoryInternalImpl.java:[50,9] cannot find symbol
  symbol:   class UserRowMapper
  location: class com.jhipster.demo.blog.repository.BlogRepositoryInternalImpl
[INFO] 3 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

##### **Motivation for or Use Case**

I like to show off new features in JHipster. ;)

##### **Reproduce the error**

Create an app with the following JDL, then run `./mvnw`.

```
application {
  config {
    baseName blog
    applicationType monolith
    authenticationType oauth2
    packageName com.jhipster.demo.blog
    prodDatabaseType mysql
    cacheProvider hazelcast
    buildTool maven
    clientFramework vue
    reactive true
    testFrameworks [cypress]
  }
  entities *
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

paginate Post, Tag with infinite-scroll
```

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

It looks like the `UserRowMapper` is missing, so maybe it just needs to be added?

![image](https://user-images.githubusercontent.com/17892/104263969-e10bf300-5447-11eb-8a68-9e773322772b.png)

##### **JHipster Version(s)**

7.0 beta 0

--- a/generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs
@@ -389,7 +389,7 @@ relationships.forEach((relationship, idx) => {
 
     <%_ if (databaseType === 'sql' && reactive) { _%>
     @Column("<%= getColumnName(relationshipName) %>_id")
-    private Long <%= relationshipFieldName %>Id;
+    private <% if (relationshipFieldName === 'user' && authenticationType === 'oauth2') { %>String<% } else { %>Long<% } %> <%= relationshipFieldName %>Id;
     <%_ } _%>
 
 <%_ } else if (relationshipType === 'many-to-many') {

--- a/generators/server/files.js
+++ b/generators/server/files.js
@@ -1078,7 +1078,14 @@ const serverFiles = {
             ],
         },
         {
+<<<<<<< HEAD
             condition: generator => generator.databaseType === 'sql' && generator.reactive && !generator.skipUserManagement,
+=======
+            condition: generator =>
+                generator.databaseType === 'sql' &&
+                generator.reactive &&
+                (!generator.skipUserManagement || generator.authenticationType === 'oauth2'),
+>>>>>>> 9756f77f79... Fix formatting
             path: SERVER_MAIN_SRC_DIR,
             templates: [
                 {

--- a/generators/server/files.js
+++ b/generators/server/files.js
@@ -1078,14 +1078,10 @@ const serverFiles = {
             ],
         },
         {
-<<<<<<< HEAD
-            condition: generator => generator.databaseType === 'sql' && generator.reactive && !generator.skipUserManagement,
-=======
             condition: generator =>
                 generator.databaseType === 'sql' &&
                 generator.reactive &&
                 (!generator.skipUserManagement || generator.authenticationType === 'oauth2'),
->>>>>>> 9756f77f79... Fix formatting
             path: SERVER_MAIN_SRC_DIR,
             templates: [
                 {

--- a/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
@@ -365,17 +365,20 @@ class UserSqlHelper {
         List<Expression> columns = new ArrayList<>();
         columns.add(Column.aliased("id", table, columnPrefix + "_id"));
         columns.add(Column.aliased("login", table, columnPrefix + "_login"));
+        <%_ if (authenticationType !== 'oauth2') { _%>
         columns.add(Column.aliased("password_hash", table, columnPrefix + "_password"));
+        <%_ } _%>
         columns.add(Column.aliased("first_name", table, columnPrefix + "_first_name"));
         columns.add(Column.aliased("last_name", table, columnPrefix + "_last_name"));
         columns.add(Column.aliased("email", table, columnPrefix + "_email"));
         columns.add(Column.aliased("activated", table, columnPrefix + "_activated"));
         columns.add(Column.aliased("lang_key", table, columnPrefix + "_lang_key"));
         columns.add(Column.aliased("image_url", table, columnPrefix + "_image_url"));
+        <%_ if (authenticationType !== 'oauth2') { _%>
         columns.add(Column.aliased("activation_key", table, columnPrefix + "_activation_key"));
         columns.add(Column.aliased("reset_key", table, columnPrefix + "_reset_key"));
         columns.add(Column.aliased("reset_date", table, columnPrefix + "_reset_date"));
-
+        <%_ } _%>
         return columns;
     }
 }

